### PR TITLE
event page refreshes properly

### PR DESCRIPTION
**Pull request by t-rutten**
params.js
- add code field to 404 error object
- alter formatting

PostPage.jsx
- single post passed in props instead of list of props
- basic 404 error handling; still work to be done here
- PostContainer renders depending on loaded bool from props AND whether postData exists

PostChangeContainer.js
- pass single post under consideration instead of mulitple posts to PostPage
- pass errors to PostPage

### CODE REVIEW

File: app/components/post/PostPage.jsx [hunk 1]
```diff
@@ -10,28 +10,34 @@ import PostContainer from './PostContainer';
...
+    if (!(!!this.props.post) && this.props.loaded) {
+      this.props.getPostData(id);
+    }
+  }
+
+  render() {
+
     const id = this.props.match.params.event_id;
-    const postData = this.props.posts[id];
+    const postData = this.props.post;
```
Comment (tgoodwin) [inline app/components/post/PostPage.jsx]: 🙌

File: app/components/post/PostPage.jsx [hunk 2]
```diff
@@ -10,28 +10,34 @@ import PostContainer from './PostContainer';
...
 
-    // TODO: ACTUAL 404 HANDLING -- in some global reduxy way!
-    // should not be handled at the component level
-    // in Main.jsx we can have a check if some
-    // this.props.404 = true, redirect to 404 page
-    if (!postData) {
-        return (<Redirect to='/' />);
+    const error = this.props.error
+    if (Object.keys(error).length > 0 && error.code === 404) {
+      return <h1> Post not found </h1>
```
Comment (tgoodwin) [inline app/components/post/PostPage.jsx]: maybe leave a comment like "TODO: actual 404 page"

File: api/middleware/params.js [hunk 3]
```diff
@@ -25,10 +25,10 @@ module.exports = (router) => {
   router.param('username', (req, res, next, username) => {
     UserService.getUserByUsername(username).then(result => {
       if (result.length === 0) {
-        return res.status(404).send({error: 'user not found'})
+        return res.status(404).send({ error: 'user not found' })
       }
       req.pathUser = result[0];
       return next();
     }).catch(err => next(err));
-  });
```
Comment (westonjackson) [inline api/middleware/params.js]: nice

File: app/components/post/PostPage.jsx (hunk 1 above)
Comment (westonjackson) [inline app/components/post/PostPage.jsx]: ???

File: app/components/post/PostPage.jsx (hunk 2 above)
Comment (westonjackson) [inline app/components/post/PostPage.jsx]: come on

File: app/components/post/PostPage.jsx [hunk 4]
```diff
@@ -10,28 +10,34 @@ import PostContainer from './PostContainer';
...
-    // this.props.getPostData(id);
-}
-addPost = () => {
+    if (!(!!this.props.post) && this.props.loaded) {
+      this.props.getPostData(id);
+    }
+  }
+
+  render() {
+
```
Comment (westonjackson) [inline app/components/post/PostPage.jsx]: nit: whitespace

File: app/components/post/PostPageContainer.js [hunk 5]
```diff
@@ -1,21 +1,23 @@
...
-import {getPostData } from '../../actions/post_actions';
+import { getPostData } from '../../actions/post_actions';
 
 import PostPage from './PostPage';
 
 const mapStateToProps = (state, ownProps) => ({
-	loggedIn: !!state.session.currentUser,
-	loaded: state.ui.loading.postsLoading === false,
-	posts: state.entities.posts
+  loggedIn: !!state.session.currentUser,
```
Comment (westonjackson) [inline app/components/post/PostPageContainer.js]: huh?

File: app/components/post/PostPage.jsx [hunk 6]
```diff
@@ -10,28 +10,34 @@ import PostContainer from './PostContainer';
...
     }
-}
-componentDidMount() {
+  }
+  componentDidMount() {
     const id = this.props.match.params.event_id;
-    // this.props.getPostData(id);
-}
-addPost = () => {
+    if (!(!!this.props.post) && this.props.loaded) {
```
Comment (tgoodwin) [inline app/components/post/PostPage.jsx]: when bool logic gets complex i sometimes like to do it like this:
```
const shouldGetPostData = !this.props.post && this.props.loaded;
if (shouldGetPostData) { ... do some shit ... }
```

File: app/components/post/PostPage.jsx (hunk 2 above)
Comment (t-rutten) [reply app/components/post/PostPage.jsx]: 👍

File: api/middleware/params.js [hunk 7]
```diff
@@ -25,10 +25,10 @@ module.exports = (router) => {
   router.param('username', (req, res, next, username) => {
     UserService.getUserByUsername(username).then(result => {
       if (result.length === 0) {
-        return res.status(404).send({error: 'user not found'})
+        return res.status(404).send({ error: 'user not found' })
```
Comment (westonjackson) [inline api/middleware/params.js]: 👎

File: app/components/post/PostPage.jsx (hunk 6 above)
Comment (t-rutten) [reply app/components/post/PostPage.jsx]: that's more readable, I'll make that change

Comment (westonjackson) [reply app/components/post/PostPage.jsx]: looks good to me

**Pull request merged**